spec/vcr_cassettes/manageiq/providers/openshift/container_manager/test_objects_record.sh

```diff
@@ -0,0 +1,111 @@
+SPEC=./spec/models/manageiq/providers/openshift/container_manager/refresher_spec.rb
+
+echo; echo "===== Clean slate, create objects ====="
+
+oc delete --ignore-not-found project my-project-{0,1,2}
+
+while oc get --show-all projects | grep my-project; do
+  echo "... waiting for projects to disappear ..."
+  sleep 3
+done
+
+for ind in 0 1 2; do
+  oc new-project my-project-$ind
+  oc project my-project-$ind
+  # Some objects here require the admin priviledges we got above.
+  oc process -f "$VCR_DIR"/test_objects_template.yml -v INDEX=$ind | oc create -f -
+
+  oc start-build my-build-config-$ind
+done
+
+while OUT="$(oc get build --all-namespaces)"; echo "$OUT"; [ "$(echo "$OUT" | egrep --count --word 'Complete|Failed|Error')" -ne 3 ]; do
+  echo "... waiting for builds to complete ..."
+  sleep 3
+done
+
+echo; echo "===== Record first VCR ====="
+
+describe_vcr () {
+  echo "VCR generated by $0 against $OPENSHIFT_MASTER_HOST"
+  echo
+  oc version
+  echo
+  echo "CAVEAT: status shown here might differ from moment captured in VCR!"
+  echo "== oc get projects --show-all --show-kind --show-labels =="
+  oc get projects --show-all --show-kind --show-labels
+  echo
+  echo "== oc get all --show-all --all-namespaces -o wide --show-labels =="
+  oc get all --show-all --all-namespaces -o wide --show-labels
+  echo
+  echo "== oc get images =="
+  oc get images
+}
+
+# Deleting VCR file allows using :new_episodes so multiple specs calling refresh
+# is not a problem, only the first will re-record the VCR.
+rm -v "$VCR_DIR"/refresher_before_deletions.{yml,txt} || true
+describe_vcr > "$VCR_DIR"/refresher_before_deletions.txt
+env RECORD_VCR=before_deletions bundle exec rspec "$SPEC" || echo "^^ FAILURES ARE POSSIBLE, YOU'LL HAVE TO EDIT THE SPEC"
+
+echo; echo "===== Various deletions ====="
+
+oc delete project my-project-0
+
+oc project my-project-1
+oc delete pod my-pod-1
+oc delete service my-service-1
+oc delete route my-route-1
+oc delete resourceQuota my-resource-quota-1
+oc delete limitRange my-limit-range-1
+oc delete persistentVolumeClaim my-persistentvolumeclaim-1
+oc delete template my-template-1
+oc delete buildconfig my-build-config-1 # also deletes its build(s)
+oc delete rc my-replicationcontroller-1
+echo "-- What remained in my-project-1 --"
+oc get all
+
+oc project my-project-2
+oc label route my-route-2 key-route-label-
+# Remove template parameters ("json merge" mode completely replaces arrays)
+oc patch --type=merge template my-template-2 --patch='{"parameters": []}'
+oc delete pod my-pod-2
+
+
+while oc get --show-all projects | grep my-project-0; do
+  echo "... waiting for my-project-0 to disappear ..."
+  sleep 3
+done
+
+echo; echo "===== Record second VCR ====="
+
+rm -v "$VCR_DIR"/refresher_after_deletions.{yml,txt} || true
+describe_vcr > "$VCR_DIR"/refresher_after_deletions.txt
+env RECORD_VCR=after_deletions bundle exec rspec "$SPEC" || echo "^^ FAILURES ARE POSSIBLE, YOU'LL HAVE TO EDIT THE SPEC"
```

Comment:
Are FAILURES expected due to the status issue above? something else?

Reply:
Due to recordings not being 100% reproducible — it's quite stable now but still usually at least something is off by one.  "An ideal world is left as an exercise to the reader."

Additionally, if you switch Openshift versions, or change template deliberately, you'll get certain differences.

In any case, this script should not abort on test fail — it should let you get both recordings, and then you can edit & run spec again and again...

Better wording welcome.